src/app/clusters/ota-server/ota-server.cpp

```diff
@@ -82,5 +141,31 @@ bool emberAfOtaSoftwareUpdateServerClusterQueryImageCallback(
     /* TYPE WARNING: array array defaults to */ uint8_t * protocolsSupported, uint8_t * location, uint8_t clientCanConsent,
     chip::ByteSpan metadataForServer)
 {
-    return false;
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    ChipLogDetail(Zcl, "OTA Server received QueryImage");
+
+    const uint8_t locationLen = emberAfStringLength(location);
+    if ((locationLen != kLocationParamLength) || (metadataForServer.size() > kMaxMetadataLen))
+    {
+        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_FIELD);
+    }
+
+    chip::ByteSpan locationSpan(location, locationLen);
+
+    if (SendStatusIfDelegateNull())
+        return true;
+    status = gServerDelegate->HandleQueryImage(vendorId, productId, imageType, hardwareVersion, currentVersion, protocolsSupported,
+                                               locationSpan, clientCanConsent, metadataForServer);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfSendImmediateDefaultResponse(status);
+    }
+
+    return true;
+}
+
+void chip::app::clusters::OTAServer::SetDelegate(OTAServerDelegate * delegate)
+{
+    gServerDelegate = delegate;
```

Comment:
Any reason why this is global? Is it because the framework doesn't allow cleanly associating particular arguments/delegates to given cluster instances?

Reply:
I don't know the answer to this, I've just seen Network Commissioning Cluster do the same. I'll see what I can find out about this

Reply:
Should we differentiate by Endpoint as well? Just in case there's an application that wants multiple OTA Server instances?